compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Output.hs

```diff
@@ -0,0 +1,232 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE DerivingStrategies #-}
+
+module DA.Daml.Doc.Render.Output
+  ( renderModule
+  ) where
+
+import DA.Daml.Doc.Types
+import DA.Daml.Doc.Render.Util (wrapOp)
+import DA.Daml.Doc.Render.Monoid
+
+import Data.List.Extra
+import Data.Maybe
+import qualified Data.Text as T
+
+renderModule :: ModuleDoc -> RenderOut
+renderModule = renderDoc
+
+class RenderDoc t where
+    renderDoc :: t -> RenderOut
+
+instance RenderDoc Anchor where renderDoc = RenderAnchor
+instance RenderDoc DocText where renderDoc = RenderDocs
+instance RenderDoc t => RenderDoc (Maybe t) where
+    renderDoc = maybe mempty renderDoc
+instance RenderDoc t => RenderDoc [t] where
+    renderDoc = mconcatMap renderDoc
+
+isModuleEmpty :: ModuleDoc -> Bool
+isModuleEmpty ModuleDoc{..} =
+    null md_templates && null md_classes &&
+    null md_adts && null md_functions &&
+    isNothing md_descr
+
+instance RenderDoc ModuleDoc where
+    renderDoc m | isModuleEmpty m = mempty
+    renderDoc ModuleDoc{..} = mconcat
+        [ renderDoc md_anchor
+        , RenderModuleHeader ("Module " <> unModulename md_name)
+        , renderDoc md_descr
+        , section "Templates" md_templates
+        , section "Template Instances" md_templateInstances
```

Comment:
`Template Instances` vs `Data types` seems weird. I would either capitalize the second word in both or neither.

Reply:
`Data Types`, as it is a title?

Reply:
👍